### ISSUE TITLE
[BUG]: `github_issue_labels` can't clobber existing labels

### ISSUE THREAD

**Issue by ezzatron** [integrations/terraform-provider-github#2089]
### Expected Behavior

I expected the `github_issue_labels` resource to allow me to specify the full list of issue labels I want, even if those issue labels overlap some of the default ones created in a new GitHub repo, or include other existing labels in the repo.

### Actual Behavior

When using `github_issue_labels` with a set of labels that includes an existing label, the apply fails with an error like:

`{"@level":"error","@message":"Error: POST https://api.github.com/repos/org/repo/labels: 422 Validation Failed [{Resource:Label Field:name Code:already_exists Message:}]","@module":"terraform.ui","@timestamp":"2024-01-10T04:48:36.512181Z","diagnostic":{"severity":"error","summary":"POST https://api.github.com/repos/org/repo/labels: 422 Validation Failed [{Resource:Label Field:name Code:already_exists Message:}]","detail":"","address":"module.name.github_issue_labels.this","range":{"filename":"modules/repo/issue-labels.tf","start":{"line":8,"column":39,"byte":246},"end":{"line":8,"column":40,"byte":247}},"snippet":{"context":"resource \"github_issue_labels\" \"this\"","code":"resource \"github_issue_labels\" \"this\" {","start_line":8,"highlight_start_offset":38,"highlight_end_offset":39,"values":[]}},"type":"diagnostic"}`

This makes it impossible to use this resource with a set of labels that includes any of the default labels for new repos. This would be fine if you could simply omit those labels, but if you do, the default labels will be removed on subsequent runs.

### Terraform Version

Terraform v1.6.6
on darwin_arm64
+ provider registry.terraform.io/integrations/github v5.43.0

### Affected Resource(s)

- github_issue_labels

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ezzatron** [comment]
As far as I'm aware this is still an issue.

**Comment by larsks** [comment]
I just ran into this myself; it makes the provider just about useless. We've had to switch to using `github_issue_label` instead, which while it works means we're now managing individual labels rather than an exclusive set.

I see there's been a pull request in place since October that looks like it addresses the problem; it would be awesome if that could get merged.

Thanks @stevehipwell for tackling this.

**Comment by stevehipwell** [comment]
@fatbasstard #2430 fixes this and adds tests (5+ months ago) but hasn't even been acknowledged by the maintainers.

**Comment by fatbasstard** [comment]
@kfcampbell @nickfloyd Do you have any time to take a look at this issue and related PR https://github.com/integrations/terraform-provider-github/pull/2430

**Comment by esacteksab** [comment]
Came here looking for this. If you create a repository, a default set of labels exist. I wish `github_repository` had a parameter to not create labels, because I also use `github_issue_labels` to define labels, but this fails every single time I create a new repository and I'm either stuck importing the default labels to only destroy or deleting the default labels so I can then apply cleanly. Tangentially, if I use a template repository, why am I not getting that template's labels? This would still be less than ideal than a repository coming up with no labels by default, but at least with taking the labels from the template repository, I'm more aligned with my default set of labels.

**Comment by ezzatron** [comment]
The workaround I'm currently using is to remove the default labels at the organization level, so that new repos created by Terraform get no labels at all. Then I'm free to set whatever labels I want with `github_issue_labels`.

Unfortunately this only works for organizations, as far as I know: https://docs.github.com/en/organizations/managing-organization-settings/managing-default-labels-for-repositories-in-your-organization#deleting-a-default-label

**Comment by esacteksab** [comment]
Yeah, I'm seeing similar behavior. More aligned with expectations. No change in the provider (v6.6.0).

**Comment by NickLarsenNZ** [comment]
I've imported the labels for a repository, but the _apply_ wants to first delete them, and then create them (exactly the same as they are).

Slightly different to the original issue description but could it be related or should I open a new issue?

**Comment by stevehipwell** [comment]
@NickLarsenNZ which provider version and could you please share the output?
